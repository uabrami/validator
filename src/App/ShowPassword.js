import React from "react";
import "./styles.css";

const ShowPassword = ({ handleToggle }) => (
  <div className="showPasswordContainer">
    <input
      className="checkbox"
      type="checkbox"
      onClick={handleToggle}
    />
    <span className="checkboxTitle">Show</span>
  </div>
);

export default ShowPassword;