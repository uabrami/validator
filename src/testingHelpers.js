import { render } from "@testing-library/react";

export const checkBasicRendering = (component) => {
  it("Expect to render", () => {
    const { container } = render(component);
    expect(container.firstChild).toBeDefined();
  });
};

export const checkRequiredProps = (component) => {
  it("Expect to have all required propTypes", () => {
    const spy = jest.spyOn(global.console, "error");
    render(component);
    expect(spy).not.toHaveBeenCalled();
    spy.mockClear();
  });
};
