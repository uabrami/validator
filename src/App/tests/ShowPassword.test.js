import React from "react";
import { render } from "@testing-library/react";

import ShowPassword from "../ShowPassword";

import { checkBasicRendering, checkRequiredProps } from "../../testingHelpers";

const testProps = { handleToggle: jest.fn() };

const Component = <ShowPassword {...testProps} />;

describe("<ShowPassword />", () => {
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });

  checkBasicRendering(Component);
  checkRequiredProps(Component);
});
