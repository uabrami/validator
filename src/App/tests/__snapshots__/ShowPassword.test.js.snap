// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowPassword /> Should render and match the snapshot 1`] = `
<div
  class="showPasswordContainer"
>
  <input
    class="checkbox"
    type="checkbox"
  />
  <span
    class="checkboxTitle"
  >
    Show
  </span>
</div>
`;
